src/components/KTable/KTable.vue

```diff
@@ -228,7 +239,7 @@ const props = defineProps({
    * @param {Array} options.data - Array of Objects defining column data
    */
   options: {
-    type: Object,
+    type: Object as PropType<TableOptions>,
```

Comment:
I have concerns about breaking existin implementations by making the prop types too strict. We want to avoid making users typecast as much as possible in the host app. Having to do things like 
![image](https://github.com/Kong/kongponents/assets/67973710/10bad7ba-5c20-402b-ab04-0ad8f4c06592)
everywhere is not a great DX.

Can we keep the types loose in the prop defs and have the strict types on a local copy of the prop? We can just cast to the correct type like we do for our handling of the `headers` prop:

https://github.com/Kong/kongponents/blob/main/src/components/KTable/KTable.vue#L729

Reply:
tried this new PR in khcp-ui and shared-ui, no ts errors